### PR TITLE
fix(plugin-meetings): throw error if no valid ice candidates are found

### PR DESCRIPTION
**Pull request by arun3528**
Fixes #https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-191901

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ pc.iceCandidate = (peerConnection, {remoteQualityLevel}) =>
 
       if (!validateSdp(peerConnection.sdp)) {
         setTimeout(() => {
```
Comment (adamweeks) [inline packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js:165]: Could you add a comment around why this nested timeout is necessary. It is a bit difficult to understand.

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js [hunk 2]
```diff
@@ -154,7 +154,7 @@ pc.setContentSlides = (screenPc) => {
  * @returns {RTCPeerConnection}
```
Comment (adamweeks) [inline packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js:154]: Let's update this to be more accurate:
```suggestion
 * @returns {Promise.<RTCPeerConnection>}
```

**Comment by arun3528** [comment]
Thanks for adding the JIRA @adamweeks

**Review by adamweeks — APPROVED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js [hunk 3]
```diff
@@ -163,7 +163,10 @@ pc.iceCandidate = (peerConnection, {remoteQualityLevel}) =>
 
       if (!validateSdp(peerConnection.sdp)) {
         setTimeout(() => {
-          resolve(peerConnection);
+          // peerconnection does gather ice candidate IP but in some cases due to firewall 
+          // or proxy the ice candidate does not get gathered so we need to wait and then retry 
+          // if still not valid then throw an error saying missing ice candidate
+          reject(new MediaError('No ice candidate or port found in the SDP'));
```
Comment (adamweeks) [inline packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js:169]: This is going to always reject after a timeout, we need to check again:
```suggestion
         if (validateSdp(peerConnection.sdp)) {
           resolve(peerConnection);
         } else {
           LoggerProxy.logger.error('PeerConnectionManager:index#iceCandidate --> SDP not valid after waiting.');
           reject(new MediaError('No ice candidate or port found in the SDP'));
         }
```

**Review by adamweeks — DISMISSED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js (hunk 3 above)
Comment (arun3528) [reply packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js]: you are correct

**Comment by arun3528** [comment]
@adamweeks  ready for review

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js [hunk 4]
```diff
@@ -102,6 +104,13 @@ const validateSdp = (sdp) => {
     if (mediaLine.candidates && mediaLine.candidates.length === 0) {
       LoggerProxy.logger.error('PeerConnectionManager:index#validateSdp --> iceCandidate: Ice candadate never completed');
       return false;
+    } else {
+      // Sometimes the candidates might be there but only IPV6 we need to makes sure we have IPV4
+      let hostCandidate = mediaLine.candidates.filter(candidate => candidate.type > HOST && candidate.ip.match(IPV4_REGEX));
+      if(hostCandidate.length === 0) {
+        LoggerProxy.logger.error('PeerConnectionManager:index#validateSdp --> iceCandidate: Ice candadate never completed');
```
Comment (adamweeks) [inline packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js:111]:
```suggestion
        LoggerProxy.logger.error('PeerConnectionManager:index#validateSdp --> iceCandidate: no IPV4 candidate present');
```

**Comment by adamweeks** [comment]
@arun3528 just the one change on the error message. Hope we get these tests working as well.

**Review by adamweeks — APPROVED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js [hunk 5]
```diff
@@ -98,21 +100,38 @@ const checkH264Support = (sdp) => {
...
     if (mediaLine.candidates && mediaLine.candidates.length === 0) {
       LoggerProxy.logger.error('PeerConnectionManager:index#validateSdp --> iceCandidate: Ice candadate never completed');
-      return false;
+      validSdp = validSdp && false;
+      return;
+    } else {
+      // Sometimes the candidates might be there but only IPV6 we need to makes sure we have IPV4
+      let hostCandidate = mediaLine.candidates.filter((candidate) =>{
+         return !!(candidate.type === HOST && candidate.ip.match(IPV4_REGEX))
+        } );
```
Comment (taymoork2) [inline packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js:114]: eslint hasn't formatted these files correctly, and looking at the file as a whole show's that eslint has been disabled
Remove the `eslint-disable` at the top of the file and have eslint autofix the file

**Review by taymoork2 — CHANGES_REQUESTED** [review]

**Review by taymoork2 — APPROVED** [review]
LGTM

**Pull request merged**